### DISCUSSION TITLE
Unable to check mod_rewrite status

### DISCUSSION THREAD

**Discussion by gregstah** [ddev/ddev#5838] [Q&A] [answered]
HI, I know that mod_rewrite is enabled in Apache, however but when trying to setup a CRM (https://www.espocrm.com/) it stalls because ESPO can't confirm that mod_rewrite is enabled in it's install checklist. 

Is there a way around this? I presume it's running code something like this:

`    if (in_array('mod_rewrite', $modules)) {
        echo "mod_rewrite is enabled.";
    }`

Thank you.

**Comment by rfay** [comment]
Hi @gregstah - are you using apache-fpm? If not, it can't work. But with `ddev config --webserver-type=apache-fpm` it should work as it would on any project with apache and FPM. 

However, if ESPO assumes that it only runs on Apache with mod-php (PHP linked directly into Apache) it won't work. There are a few older PHP projects that make that assumption.

@GuySartorelli has talked about doing a DDEV add-on or recipe to enable mod-php in DDEV. We had it for a few years but dropped it 4-5 years ago because it was little used and the performance was terrible.

**Reply by gregstah** [reply, marked as answer]
Hi Randy, thank you for your help. 

As you suspected, it defaults to "**nginx-fpm**" updating the yaml file to **apache-fpm** solved the problem.

It's working perfectly. I'm new to ddev and can't get over how blazingly fast it is.

**Reply by rfay** [reply]
Yay, glad it worked!
